src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -1431,18 +1432,41 @@ static void run_server(Preforker& forker, NBDServer *server, bool netlink_used)
   shutdown_async_signal_handler();
 }
 
+// Eventually it should be removed when pidfd_open is widely supported.
+
+static int wait_for_terminate_legacy(int pid, int timeout)
+{
+  for (int i = 0; ; i++) {
+    if (kill(pid, 0) == -1) {
+      if (errno == ESRCH) {
+        return 0;
+      }
+      int r = -errno;
+      cerr << "rbd-nbd: kill(" << pid << ", 0) failed: "
+           << cpp_strerror(r) << std::endl;
+      return r;
+    }
+    if (i >= timeout * 2) {
+      break;
+    }
+    usleep(500000);
+  }
+
+  cerr << "rbd-nbd: waiting for process exit timed out" << std::endl;
+  return -ETIMEDOUT;
+}
+
 // Eventually it should be replaced with glibc' pidfd_open
 // when it is widely available.
+
+#ifndef __NR_pidfd_open
+#define __NR_pidfd_open 434   /* System call # on most architectures */
```

Comment:
Nit: I still think we should avoid defining this if it doesn't exist. Instead, use `#ifdef __NR_pidfd_open` to create a stub method that just returns ENOSYS.

Reply:
I just copied this from the man page example [1]. And actually to me it makes sense: if the currently installed unistd.h header does not have the syscall, it does not mean it is not supported by the currently running kernel (or a kernel where rbd-nbd will eventually be running) and vice versa, if it is defined in the header it does not mean the kernel will support it.

So I am in favor to leave it as it is currently, but if my reasoning has not changed your mind, I will update the code.

[1] https://man7.org/linux/man-pages/man2/pidfd_open.2.html

Reply:
I personally still don't like it -- unless you perhaps wrapped it w/ a `LINUX` guard. The closest other example we have to this in Ceph is the `__NR_syncfs` syscall, but it doesn't attempt to define it.